pkg/vgmanager/vgmanager_controller.go

```diff
@@ -162,19 +163,30 @@ func (r *VGReconciler) reconcile(ctx context.Context, volumeGroup *lvmv1alpha1.L
 			}
 		}
 
-		if devicesExist {
-			r.Log.Info("all the available devices are attached to the volume group", "VGName", volumeGroup.Name)
-			if statuserr := r.setVolumeGroupReadyStatus(ctx, volumeGroup.Name); statuserr != nil {
-				r.Log.Error(statuserr, "failed to update status", "VGName", volumeGroup.Name)
+		if !devicesExist {
+			err := fmt.Errorf("no available devices found for volume group")
+			r.Log.Error(err, err.Error(), "VGName", volumeGroup.Name)
+			if statuserr := r.setVolumeGroupFailedStatus(ctx, volumeGroup.Name, err.Error()); statuserr != nil {
+				r.Log.Error(statuserr, "failed to update status", "name", volumeGroup.Name)
 				return reconcileAgain, statuserr
 			}
-		} else {
-			errMsg := "no available devices found for volume group"
-			r.Log.Error(fmt.Errorf(errMsg), errMsg, "VGName", volumeGroup.Name)
-			if statuserr := r.setVolumeGroupFailedStatus(ctx, volumeGroup.Name, errMsg); statuserr != nil {
+			return reconcileAgain, err
+		}
+
+		if err := r.validateLVs(volumeGroup); err != nil {
```

Comment:
good point, I added two oneliner comments to reflect that the validation in the readiness check is for consistency purpose while on the create of the thin pool it is to validate everything in create went as planned.

Reply:
I will add though that because this validation can fail here, there would be manual cleanup necessary because the VG / PVs could still be created even though the LVs are corrupt.

Reply:
We can try to force remove them on error here, but not sure what the best case is. What do you think?

Reply:
I think we should avoid force removing. Instead, setting the state to `Degraded` with a good explanation should be sufficient.

Reply:
Agree on that. In this case it would show failed for new creations and degraded for existing ones i believe, since for degraded there must be an active volumegroup in the status. I consider that good enough then.

Reply:
Exactly.